spec/models/user_spec.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User do
+  describe 'when have validations' do
+    it { is_expected.to validate_presence_of(:name) }
+  end
+
+  describe 'when have associations' do
+    it { is_expected.to have_many(:searchs).dependent(:destroy) }
+    it { is_expected.to have_many(:key_words).through(:searchs) }
+  end
```

Comment:
@adrianoavelino Veja que no outro teste eu coloquei o describe apenas como "validations" e "association" quando contem "with", "when", "without".... É para contextos, normalmente você descreve apenas o método, ou a parte mais genérica, nesses casos, como comentei acima.

```suggestion
  describe 'validations' do
    it { is_expected.to validate_presence_of(:name) }
  end

  describe 'associations' do
    it { is_expected.to have_many(:searchs).dependent(:destroy) }
    it { is_expected.to have_many(:key_words).through(:searchs) }
  end
```